nomos-libp2p/src/config.rs

```diff
@@ -173,6 +207,228 @@ pub mod secret_key_serde {
+    Unfiltered,
+    FilterBoth,
+}
+
+#[derive(Debug, Clone, Serialize, Deserialize, PartialEq, Eq)]
+#[serde(rename_all = "snake_case")]
+pub enum KBucketInserts {
+    OnConnected,
+    Manual,
+}
+
+#[derive(Debug, Clone, Serialize, Deserialize, PartialEq, Eq)]
+#[serde(tag = "type", content = "config")]
+pub enum CachingSettings {
+    Disabled,
+    Enabled { max_peers: u16 },
+}
+
+impl KademliaSettings {
+    #[must_use]
+    pub fn to_libp2p_config(&self, protocol_name: ProtocolName) -> kad::Config {
+        let mut config = kad::Config::new(StreamProtocol::new(protocol_name.kad_protocol_name()));
+
+        if let Some(timeout) = self.query_timeout_secs {
+            config.set_query_timeout(Duration::from_secs(timeout));
+        }
+
+        handle_non_zero(self.replication_factor, "Replication factor", |factor| {
+            config.set_replication_factor(factor);
+        });
+
+        handle_non_zero(self.parallelism, "Parallelism", |parallel| {
+            config.set_parallelism(parallel);
+        });
+
+        if let Some(disjoint) = self.disjoint_query_paths {
+            config.disjoint_query_paths(disjoint);
+        }
+
+        if let Some(size) = self.max_packet_size {
+            config.set_max_packet_size(size);
+        }
+
+        if let Some(inserts) = &self.kbucket_inserts {
+            match inserts {
+                KBucketInserts::OnConnected => {
+                    config.set_kbucket_inserts(kad::BucketInserts::OnConnected);
+                }
+                KBucketInserts::Manual => {
+                    config.set_kbucket_inserts(kad::BucketInserts::Manual);
+                }
+            }
+        }
+
+        if let Some(caching) = &self.caching {
+            match caching {
+                CachingSettings::Disabled => {
+                    config.set_caching(kad::Caching::Disabled);
+                }
+                CachingSettings::Enabled { max_peers } => {
+                    config.set_caching(kad::Caching::Enabled {
+                        max_peers: *max_peers,
+                    });
+                }
+            }
+        }
+
+        // Set the periodic bootstrap interval if provided
+        // If not provided, use the libp2p default
+        // disable periodic bootstrap if interval is 0
+        if let Some(interval) = &self.periodic_bootstrap_interval_secs {
+            if *interval == 0 {
+                config.set_periodic_bootstrap_interval(None);
+            } else {
+                config.set_periodic_bootstrap_interval(Some(Duration::from_secs(*interval)));
+            }
+        }
+
+        config
+    }
+}
+
+fn handle_non_zero<F>(value: Option<usize>, field_name: &str, f: F)
+where
+    F: FnOnce(NonZeroUsize),
+{
+    if let Some(val) = value {
+        if let Some(non_zero) = NonZeroUsize::new(val) {
+            f(non_zero);
+        } else {
+            tracing::warn!("{} value is 0, setting to default", field_name);
+        }
+    }
+}
```

Comment:
This is what I sad can be replace by:`Option<NonZero<usize>>` during serialization. Forcing the user to set something valid or nothing for default.

Reply:
https://github.com/logos-co/nomos/pull/1154/commits/6ceb44ec3bca2380de5a87992946ac61c9bf072d. That seems much better, thanks!